framework/src/meshgenerators/ElementDeletionGeneratorBase.C

```diff
@@ -52,6 +52,16 @@ ElementDeletionGeneratorBase::generate()
       deleteable_elems.insert(elem);
   }
 
+  // Set to be merged into deleteable elems, used to avoid derefencing a dangling pointer
+  std::set<Elem *> deleteable_parents;
+
+  // check for dangling interior parents
+  for (auto & elem : mesh->element_ptr_range())
+    if (elem->dim() == mesh->mesh_dimension() - 1 &&
+        deleteable_elems.count(elem->interior_parent()) > 0)
+      deleteable_parents.insert(elem);
+
+  deleteable_elems.insert(deleteable_parents.begin(), deleteable_parents.end());
```

Comment:
why cant you insert in deleteable elements?

also isnt it deleteable_child rather than parents here?